x-pack/plugins/security_solution/public/common/components/inspect/index.tsx

```diff
@@ -6,50 +6,34 @@
  */
 
 import { EuiButtonEmpty, EuiButtonIcon } from '@elastic/eui';
-import { getOr, omit } from 'lodash/fp';
+import { omit } from 'lodash/fp';
 import React, { useCallback } from 'react';
 import { connect, ConnectedProps } from 'react-redux';
-import styled, { css } from 'styled-components';
 
 import { inputsSelectors, State } from '../../store';
 import { InputsModelId } from '../../store/inputs/constants';
 import { inputsActions } from '../../store/inputs';
 
+import { HoverVisibilityContainer } from '../hover_visibility_container';
+
 import { ModalInspectQuery } from './modal';
 import * as i18n from './translations';
 
 export const BUTTON_CLASS = 'inspectButtonComponent';
 
-export const InspectButtonContainer = styled.div<{ show?: boolean }>`
-  width: 100%;
-  display: flex;
-  flex-grow: 1;
-
-  > * {
-    max-width: 100%;
-  }
-
-  .${BUTTON_CLASS} {
-    pointer-events: none;
-    opacity: 0;
-    transition: opacity ${(props) => getOr(250, 'theme.eui.euiAnimSpeedNormal', props)} ease;
-  }
-
-  ${({ show }) =>
-    show &&
-    css`
-      &:hover .${BUTTON_CLASS} {
-        pointer-events: auto;
-        opacity: 1;
-      }
-    `}
-`;
-
-InspectButtonContainer.displayName = 'InspectButtonContainer';
+interface InspectButtonContainerProps {
+  show?: boolean;
+  children: React.ReactNode;
+}
 
-InspectButtonContainer.defaultProps = {
-  show: true,
-};
+export const InspectButtonContainer: React.FC<InspectButtonContainerProps> = ({
```

Comment:
`InspectButtonContainer` looks very similar to `HoverVisibilityContainer`, could we just keep one of them?

Reply:
Yes, the idea is that `HoverVisibilityContainer` is a generic version of `InspectButtonContainer`. The only difference is that it only works for the inspect button because `targetClassNames='inspectButtonComponent'`. Instead of changing every place using `InspectButtonContainer`, I thought that would be safer to keep `InspectButtonContainer` as it is and make it use `HoverVisibilityContainer` internally. If you think that it is a bad idea I can update every `InspectButtonContainer` call. There are only 12 places using it.

Reply:
Nah, then let's keep it as it is.